### DISCUSSION TITLE
Single File CLI with puppeteer

### DISCUSSION THREAD

**Discussion by takingurstuff** [gildas-lormeau/single-file-cli#99] [Q&A] [answered]
Hello
I am currently trying to intergrate Single File to my WIP project that aims to be able to download pages with lazy loading with Puppeteer. I would like some documentation on how to use the Single File API script I found in the project

Help appreciated in advance
Thanks

**Comment by gildas-lormeau** [comment, marked as answer]
HI,

SingleFile CLI used to rely on Puppeteer. I've removed this dependency and use CDP directly to lighten the application. The last version of SingleFile CLI that used Puppeteer is https://github.com/gildas-lormeau/single-file-cli/tree/0cb134df0638f8c7f198097d8ff830a16f8b9f9d. See here for the code specific to Puppeteer: https://github.com/gildas-lormeau/single-file-cli/blob/0cb134df0638f8c7f198097d8ff830a16f8b9f9d/back-ends/puppeteer.js.

Normally, it should still work even with `single-file-core` up to date.

**Reply by takingurstuff** [reply]
thanks for the information in the previous versions of this software that utilizes puppeteer

I have utilized the api script that was in the newest version of the repo to launch a browser instance controllable by puppeteer, but i am missing components i need to finish the downloading aspect of my project, but i cannot find any relevant information regarding the API. 

I need to know how to actually save a page with the api and where the software saves the captured HTML pages.

Thanks in advance

**Reply by gildas-lormeau** [reply]
I would recommend you to clone the version I mentioned and debug the code. This would probably help you to understand how the integration with puppeteer works.
